test/coder_junit.go

```diff
@@ -0,0 +1,52 @@
+package test
+
+import (
+	"encoding/json"
+	"io"
+	"strconv"
+
+	"github.com/coduno/api/model"
+	"github.com/coduno/api/runner"
+	"github.com/coduno/api/util"
+	"github.com/coduno/api/ws"
+	"golang.org/x/net/context"
+)
+
+func init() {
+	RegisterTester(CoderJunit, coderJunit)
+}
+
+func coderJunit(ctx context.Context, t model.KeyedTest, sub model.KeyedSubmission, ball io.Reader) (err error) {
+	if _, ok := t.Params["test"]; !ok {
+		return ErrMissingParam("code")
+	}
+
+	code := model.StoredObject{
+		Bucket: util.TestsBucket,
+		Name:   t.Params["test"],
+	}
+	codeStream := stream(ctx, code)
+	var tr *model.JunitTestResult
+	if tr, err = runner.JUnit(ctx, ball, codeStream); err != nil {
+		return
+	}
+
+	shouldFail, err := strconv.ParseBool(t.Params["shouldFail"])
+	if err != nil {
+		return
+	}
+	ctr := model.CoderJunitTestResult{
+		JunitTestResult: *tr,
+		ExpectedFailure: shouldFail,
```

Comment:
LOL. I did not see this when I wrote my earlier comment. I think that `ShouldFail` is WAY better than `ExpectedFailure`.